CI: pin golangci-lint to latest tag instead of master

The CI lint job pulls golangci-lint's installer from the master branch with no version pin, which broke on 2026-04-25 when an upstream change made the default version resolve to an empty string.

## Repro

PR #2 (mergeable, all tests green locally and in CI) failed the lint job with:

```
golangci/golangci-lint info checking GitHub for latest tag
golangci/golangci-lint crit unable to find '' - use 'latest' or see https://github.com/golangci/golangci-lint/releases for details
##[error]Process completed with exit code 1.
```

Run: https://github.com/JLugagne/go-surgeon/actions/runs/24941091999

The previous run on `main` (commit `fdedd45`, ~10 min earlier) succeeded with the same workflow. The breakage is purely upstream.

## Root cause

`.github/workflows/test.yml:32`:

```yaml
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /usr/local/bin
```

Two problems:
- Pulls `install.sh` from `master` (moving target).
- Calls it with no version argument, so the script tries to resolve the default and fails when upstream tag resolution breaks.

## Proposed fix

Resolve the latest released tag at workflow runtime via the GitHub API and pass it explicitly:

```yaml
- name: Install and run golangci-lint
  run: |
    LATEST=$(curl -sSfL https://api.github.com/repos/golangci/golangci-lint/releases/latest | grep -o '"tag_name": *"[^"]*"' | head -1 | cut -d'"' -f4)
    echo "Installing golangci-lint $LATEST"
    curl -sSfL "https://raw.githubusercontent.com/golangci/golangci-lint/$LATEST/install.sh" | sh -s -- -b /usr/local/bin "$LATEST"
    golangci-lint run ./...
```

This:
- Always uses the latest released tag (matches the user's intent: "always use the latest tag of golangci-lint, not master").
- Pulls `install.sh` from the same tag, so installer and binary stay in sync.
- Passes the version explicitly to `install.sh`, sidestepping its default-resolution logic.
- Logs which version was installed for debuggability.

## Acceptance

- Re-run CI on a PR after the fix → lint job succeeds against the current latest golangci-lint release.
- Workflow logs show the resolved version.
- No change to which packages get linted (`./...`).

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -29,7 +29,9 @@ jobs:
         run: go vet ./...
       - name: Install and run golangci-lint
         run: |
-          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /usr/local/bin
+          LATEST=$(curl -sSfL https://api.github.com/repos/golangci/golangci-lint/releases/latest | grep -o '"tag_name": *"[^"]*"' | head -1 | cut -d'"' -f4)
+          echo "Installing golangci-lint $LATEST"
+          curl -sSfL "https://raw.githubusercontent.com/golangci/golangci-lint/$LATEST/install.sh" | sh -s -- -b /usr/local/bin "$LATEST"
           golangci-lint run ./...
       - name: Run govulncheck
         run: |

